src/App.tsx

```diff
@@ -15,36 +16,102 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  ALPHABETICALLY = 'alphabet',
+  LENGTH = 'length',
+  DEFAULT = '',
+}
+
+interface Options {
+  goods: string[];
+  sortBy: SortType;
+  isReversed: boolean;
+}
+
+const getGoods = ({ goods, sortBy, isReversed }: Options) => {
+  let visibleGoods = [...goods];
+
+  if (sortBy) {
+    switch (sortBy) {
+      case SortType.ALPHABETICALLY:
+        visibleGoods = visibleGoods.toSorted();
+        break;
+      case SortType.LENGTH:
+        visibleGoods = visibleGoods.toSorted((a, b) => a.length - b.length);
+        break;
+      default:
+        return null;
+    }
+  }
+
+  if (isReversed) {
+    visibleGoods = visibleGoods.toReversed();
+  }
+
+  return visibleGoods;
+};
+
+export const App = () => {
+  const [sortBy, setSortBy] = useState<SortType>(SortType.DEFAULT);
+  const [isReversed, setIsReversed] = useState(false);
+  const canReset = sortBy || isReversed;
+
+  const visibleGoods =
+    getGoods({ goods: goodsFromServer, sortBy, isReversed }) || [];
```

Comment:
When all the stuff doesn't fit into one line, we usually format it like this:

```suggestion
  const visibleGoods = getGoods({ 
    goods: goodsFromServer, 
    sortBy, 
    isReversed 
  }) || [];
```

Reply:
thanks, fixed the comments above